chore: Bump pytest version

mclasmeier opened the pull request:
### Description

Bump pytest version and recreate requirements according to the README.

### User-facing documentation

- [x] CHANGELOG update is not needed
- [x] documentation PR is not needed

### Testing and quality

<!--
General Availability requirements: https://github.com/stackrox/stackrox/blob/master/PR_GA.md
Feature Flags usage: https://github.com/stackrox/stackrox/blob/master/pkg/features/README.md
-->

- [ ] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

<!--
If no tests have been contributed, please explain why unless it's obvious,
e.g., the PR is a one-line comment change.
-->

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

<!--
Use this space to explain **how you validated** that **your change functions
exactly how you expect it**. Feel free to attach JSON snippets, curl commands,
screenshots, etc. Apply a simple benchmark: would the information you provided
convince any reviewer or any external reader that you did enough to validate
your change.

It is acceptable to assume trust and keep this section light, e.g. as a
bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a
markdown or code comment change only. It is also acceptable to skip testing for
changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting,
fixing, etc. Make sure you validate the change ASAP after it gets merged or
explain in PR when the validation will be performed. Explain here why you
skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation
activities you did manually and why so.
-->

change me!

mclasmeier commented:
@tommartensen, regarding the removal of `calver` -- as I understand it, this removal is independent of my pytest version bump. I have tried this out on master, following the README instructions, the command

```
pip-compile requirements-build.in --allow-unsafe --generate-hashes
```

removes `calver` from the `requirements-build.txt` file. (You apparently don't need the stdout redirection currently described in the README.)

So, could you maybe comment on the correctness of this `calver` removal? I am not familiar with this. Does this dependecy actually end up in your `requirements-build.in` file, when you follow the steps?

tommartensen commented:
Regarding `calver`:
I have Python `3.9.20` (with pyenv) and pip-compile `7.4.1` installed. Python 3.9 because that is the version that we use on Konflux.
I need the stdout redirect for `pip-compile requirements.in --generate-hashes > requirements.txt` to see the changes in `requirements.txt`. On your PR, the command would bump attrs and tomli in addition to the calver removal. 

`requirements-build.in` has the same content as your PR and `pip-compile requirements-build.in --allow-unsafe --generate-hashes` updates the the `requirements-build.txt` to include the calver dependency. 

Can you check your Python and pip-compile versions?

tommartensen commented:
Diff: 

```bash
diff --git a/operator/bundle_helpers/requirements-build.txt b/operator/bundle_helpers/requirements-build.txt
index d5ff4ef079..9c508598ca 100644
--- a/operator/bundle_helpers/requirements-build.txt
+++ b/operator/bundle_helpers/requirements-build.txt
@@ -4,6 +4,10 @@
 #
 #    pip-compile --allow-unsafe --generate-hashes requirements-build.in
 #
+calver==2022.6.26 \
+    --hash=sha256:a1d7fcdd67797afc52ee36ffb8c8adf6643173864306547bfd1380cbce6310a0 \
+    --hash=sha256:e05493a3b17517ef1748fbe610da11f10485faa7c416b9d33fd4a52d74894f8b
+    # via -r requirements-build.in
 cython==3.0.11 \
     --hash=sha256:0b1d1f6f94cc5d42a4591f6d60d616786b9cd15576b112bc92a23131fcf38020 \
     --hash=sha256:0e25f6425ad4a700d7f77cd468da9161e63658837d1bc34861a9861a4ef6346d \
@@ -116,7 +120,7 @@ pluggy==1.5.0 \
     # via
     #   -r requirements-build.in
     #   hatchling
-setuptools-scm==8.1.0 \
+setuptools-scm[toml]==8.1.0 \
     --hash=sha256:42dea1b65771cba93b7a515d65a65d8246e560768a66b9106a592c8e7f26c8a7 \
     --hash=sha256:897a3226a6fd4a6eb2f068745e49733261a21f70b1bb28fce0339feb978d9af3
     # via
diff --git a/operator/bundle_helpers/requirements.txt b/operator/bundle_helpers/requirements.txt
index aa12880ad5..070664ed49 100644
--- a/operator/bundle_helpers/requirements.txt
+++ b/operator/bundle_helpers/requirements.txt
@@ -4,9 +4,9 @@
 #
 #    pip-compile --generate-hashes requirements.in
 #
-attrs==24.2.0 \
-    --hash=sha256:5cfb1b9148b5b086569baec03f20d7b6bf3bcacc9a42bebf87ffaaca362f6346 \
-    --hash=sha256:81921eb96de3191c8258c199618104dd27ac608d9366f5e35d011eae1867ede2
+attrs==24.3.0 \
+    --hash=sha256:8f5c07333d543103541ba7be0e2ce16eeee8130cb0b3f9238ab904ce1e85baff \
+    --hash=sha256:ac96cd038792094f438ad1f6ff80837353805ac950cd2aa0e0625ef19850c308
     # via pytest
 exceptiongroup==1.2.2 \
     --hash=sha256:3111b9d131c238bec2f8f516e123e14ba243563fb135d3fe885990585aa7795b \
@@ -83,7 +83,37 @@ pyyaml==6.0.2 \
     --hash=sha256:f753120cb8181e736c57ef7636e83f31b9c0d1722c516f7e86cf15b7aa57ff12 \
     --hash=sha256:ff3824dc5261f50c9b0dfb3be22b4567a6f938ccce4587b38952d85fd9e9afe4
     # via -r requirements.in
-tomli==2.0.2 \
-    --hash=sha256:2ebe24485c53d303f690b0ec092806a085f07af5a5aa1464f3931eec36caaa38 \
-    --hash=sha256:d46d457a85337051c36524bc5349dd91b1877838e2979ac5ced3e710ed8a60ed
+tomli==2.2.1 \
+    --hash=sha256:023aa114dd824ade0100497eb2318602af309e5a55595f76b626d6d9f3b7b0a6 \
+    --hash=sha256:02abe224de6ae62c19f090f68da4e27b10af2b93213d36cf44e6e1c5abd19fdd \
+    --hash=sha256:286f0ca2ffeeb5b9bd4fcc8d6c330534323ec51b2f52da063b11c502da16f30c \
+    --hash=sha256:2d0f2fdd22b02c6d81637a3c95f8cd77f995846af7414c5c4b8d0545afa1bc4b \
+    --hash=sha256:33580bccab0338d00994d7f16f4c4ec25b776af3ffaac1ed74e0b3fc95e885a8 \
+    --hash=sha256:400e720fe168c0f8521520190686ef8ef033fb19fc493da09779e592861b78c6 \
+    --hash=sha256:40741994320b232529c802f8bc86da4e1aa9f413db394617b9a256ae0f9a7f77 \
+    --hash=sha256:465af0e0875402f1d226519c9904f37254b3045fc5084697cefb9bdde1ff99ff \
+    --hash=sha256:4a8f6e44de52d5e6c657c9fe83b562f5f4256d8ebbfe4ff922c495620a7f6cea \
+    --hash=sha256:4e340144ad7ae1533cb897d406382b4b6fede8890a03738ff1683af800d54192 \
+    --hash=sha256:678e4fa69e4575eb77d103de3df8a895e1591b48e740211bd1067378c69e8249 \
+    --hash=sha256:6972ca9c9cc9f0acaa56a8ca1ff51e7af152a9f87fb64623e31d5c83700080ee \
+    --hash=sha256:7fc04e92e1d624a4a63c76474610238576942d6b8950a2d7f908a340494e67e4 \
+    --hash=sha256:889f80ef92701b9dbb224e49ec87c645ce5df3fa2cc548664eb8a25e03127a98 \
+    --hash=sha256:8d57ca8095a641b8237d5b079147646153d22552f1c637fd3ba7f4b0b29167a8 \
+    --hash=sha256:8dd28b3e155b80f4d54beb40a441d366adcfe740969820caf156c019fb5c7ec4 \
+    --hash=sha256:9316dc65bed1684c9a98ee68759ceaed29d229e985297003e494aa825ebb0281 \
+    --hash=sha256:a198f10c4d1b1375d7687bc25294306e551bf1abfa4eace6650070a5c1ae2744 \
+    --hash=sha256:a38aa0308e754b0e3c67e344754dff64999ff9b513e691d0e786265c93583c69 \
+    --hash=sha256:a92ef1a44547e894e2a17d24e7557a5e85a9e1d0048b0b5e7541f76c5032cb13 \
+    --hash=sha256:ac065718db92ca818f8d6141b5f66369833d4a80a9d74435a268c52bdfa73140 \
+    --hash=sha256:b82ebccc8c8a36f2094e969560a1b836758481f3dc360ce9a3277c65f374285e \
+    --hash=sha256:c954d2250168d28797dd4e3ac5cf812a406cd5a92674ee4c8f123c889786aa8e \
+    --hash=sha256:cb55c73c5f4408779d0cf3eef9f762b9c9f147a77de7b258bef0a5628adc85cc \
+    --hash=sha256:cd45e1dc79c835ce60f7404ec8119f2eb06d38b1deba146f07ced3bbc44505ff \
+    --hash=sha256:d3f5614314d758649ab2ab3a62d4f2004c825922f9e370b29416484086b264ec \
+    --hash=sha256:d920f33822747519673ee656a4b6ac33e382eca9d331c87770faa3eef562aeb2 \
+    --hash=sha256:db2b95f9de79181805df90bedc5a5ab4c165e6ec3fe99f970d0e302f384ad222 \
+    --hash=sha256:e59e304978767a54663af13c07b3d1af22ddee3bb2fb0618ca1593e4f593a106 \
+    --hash=sha256:e85e99945e688e32d5a35c1ff38ed0b3f41f43fad8df0bdf79f72b2ba7bc5272 \
+    --hash=sha256:ece47d672db52ac607a3d9599a9d48dcb2f2f735c6c2d1f34130085bb12b112a \
+    --hash=sha256:f4039b9cbc3048b2416cc57ab3bda989a6fcf9b36cf8937f01a6e731b64f80d7
     # via pytest

```

mclasmeier commented:
This seems odd to me.

```
stackrox on  mc/operator-bundle-helpers-version-update [$?] via 🐹 v1.22.5 on ☁️  mclasmei@redhat.com 
❯ nix-shell -p python39 python39Packages.pip-tools

stackrox on  mc/operator-bundle-helpers-version-update [$?] via 🐹 v1.22.5 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ python3 --version
Python 3.9.19

stackrox on  mc/operator-bundle-helpers-version-update [$?] via 🐹 v1.22.5 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ python --version
Python 3.9.19

stackrox on  mc/operator-bundle-helpers-version-update [$?] via 🐹 v1.22.5 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ pip-compile --version
pip-compile, version 7.4.1

stackrox on  mc/operator-bundle-helpers-version-update [$?] via 🐹 v1.22.5 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ git switch master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

stackrox on  master [$?] via 🐹 v1.22.5 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ cd operator/bundle_helpers/

stackrox/operator/bundle_helpers on  master [$?] via 🐍 v3.9.19 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ git st .
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

stackrox/operator/bundle_helpers on  master [$?] via 🐍 v3.9.19 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ ./pip_find_builddeps.py requirements.txt   --append   --only-write-on-update   -o requirements-build.in
INFO: Please make sure the input files meet the requirements of this script (see --help)
INFO: Running pip download, this may take a while
INFO: Looking for build dependencies in the output of pip download
INFO: No new build dependencies found.

stackrox/operator/bundle_helpers on  master [$?] via 🐍 v3.9.19 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com took 14s 
❯ git st .
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

stackrox/operator/bundle_helpers on  master [$?] via 🐍 v3.9.19 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ pip-compile requirements-build.in --allow-unsafe --generate-hashes
  WARNING: The candidate selected for download or install is a yanked version: 'hatchling' candidate (version 1.26.1 at https://files.pythonhosted.org/packages/df/69/86bff19bf75d4f5c8bfb7597e6799ea7764a6a1378caabde8fe998fc819e/hatchling-1.26.1-py3-none-any.whl (from https://pypi.org/simple/hatchling/) (requires-python:>=3.8))
  Reason for being yanked: Upload issues
WARNING: --strip-extras is becoming the default in version 8.0.0. To silence this warning, either use --strip-extras to opt into the new default or use --no-strip-extras to retain the existing behavior.
#
# This file is autogenerated by pip-compile with Python 3.9
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes requirements-build.in
#
cython==3.0.11 \
    --hash=sha256:0b1d1f6f94cc5d42a4591f6d60d616786b9cd15576b112bc92a23131fcf38020 \
    --hash=sha256:0e25f6425ad4a700d7f77cd468da9161e63658837d1bc34861a9861a4ef6346d \
    --hash=sha256:0fc6fdd6fa493be7bdda22355689d5446ac944cd71286f6f44a14b0d67ee3ff5 \
    --hash=sha256:104d6f2f2c827ccc5e9e42c80ef6773a6aa94752fe6bc5b24a4eab4306fb7f07 \
    --hash=sha256:11996c40c32abf843ba652a6d53cb15944c88d91f91fc4e6f0028f5df8a8f8a1 \
    --hash=sha256:13062ce556a1e98d2821f7a0253b50569fdc98c36efd6653a65b21e3f8bbbf5f \
    --hash=sha256:14701edb3107a5d9305a82d9d646c4f28bfecbba74b26cc1ee2f4be08f602057 \
    --hash=sha256:187685e25e037320cae513b8cc4bf9dbc4465c037051aede509cbbf207524de2 \
    --hash=sha256:1dd47865f4c0a224da73acf83d113f93488d17624e2457dce1753acdfb1cc40c \
    --hash=sha256:221de0b48bf387f209003508e602ce839a80463522fc6f583ad3c8d5c890d2c1 \
    --hash=sha256:2252b5aa57621848e310fe7fa6f7dce5f73aa452884a183d201a8bcebfa05a00 \
    --hash=sha256:2a8ea2e7e2d3bc0d8630dafe6c4a5a89485598ff8a61885b74f8ed882597efd5 \
    --hash=sha256:301bde949b4f312a1c70e214b0c3bc51a3f955d466010d2f68eb042df36447b0 \
    --hash=sha256:3379c6521e25aa6cd7703bb7d635eaca75c0f9c7f1b0fdd6dd15a03bfac5f68d \
    --hash=sha256:351955559b37e6c98b48aecb178894c311be9d731b297782f2b78d111f0c9015 \
    --hash=sha256:3699391125ab344d8d25438074d1097d9ba0fb674d0320599316cfe7cf5f002a \
    --hash=sha256:3999fb52d3328a6a5e8c63122b0a8bd110dfcdb98dda585a3def1426b991cba7 \
    --hash=sha256:3f2b062f6df67e8a56c75e500ca330cf62c85ac26dd7fd006f07ef0f83aebfa3 \
    --hash=sha256:3ff8ac1f0ecd4f505db4ab051e58e4531f5d098b6ac03b91c3b902e8d10c67b3 \
    --hash=sha256:421017466e9260aca86823974e26e158e6358622f27c0f4da9c682f3b6d2e624 \
    --hash=sha256:4341d6a64d47112884e0bcf31e6c075268220ee4cd02223047182d4dda94d637 \
    --hash=sha256:44292aae17524abb4b70a25111fe7dec1a0ad718711d47e3786a211d5408fdaa \
    --hash=sha256:46aec30f217bdf096175a1a639203d44ac73a36fe7fa3dd06bd012e8f39eca0f \
    --hash=sha256:473d35681d9f93ce380e6a7c8feb2d65fc6333bd7117fbc62989e404e241dbb0 \
    --hash=sha256:4e9a8d92978b15a0c7ca7f98447c6c578dc8923a0941d9d172d0b077cb69c576 \
    --hash=sha256:52186101d51497519e99b60d955fd5cb3bf747c67f00d742e70ab913f1e42d31 \
    --hash=sha256:52205347e916dd65d2400b977df4c697390c3aae0e96275a438cc4ae85dadc08 \
    --hash=sha256:525d09b3405534763fa73bd78c8e51ac8264036ce4c16d37dfd1555a7da6d3a7 \
    --hash=sha256:53b6072a89049a991d07f42060f65398448365c59c9cb515c5925b9bdc9d71f8 \
    --hash=sha256:598699165cfa7c6d69513ee1bffc9e1fdd63b00b624409174c388538aa217975 \
    --hash=sha256:63f2c892e9f9c1698ecfee78205541623eb31cd3a1b682668be7ac12de94aa8e \
    --hash=sha256:6823aef13669a32caf18bbb036de56065c485d9f558551a9b55061acf9c4c27f \
    --hash=sha256:6fb68cef33684f8cc97987bee6ae919eee7e18ee6a3ad7ed9516b8386ef95ae6 \
    --hash=sha256:7146dd2af8682b4ca61331851e6aebce9fe5158e75300343f80c07ca80b1faff \
    --hash=sha256:75ba1c70b6deeaffbac123856b8d35f253da13552207aa969078611c197377e4 \
    --hash=sha256:780f89c95b8aec1e403005b3bf2f0a2afa060b3eba168c86830f079339adad89 \
    --hash=sha256:790263b74432cb997740d73665f4d8d00b9cd1cecbdd981d93591ddf993d4f12 \
    --hash=sha256:8948802e1f5677a673ea5d22a1e7e273ca5f83e7a452786ca286eebf97cee67c \
    --hash=sha256:8acdc87e9009110adbceb7569765eb0980129055cc954c62f99fe9f094c9505e \
    --hash=sha256:8b14c24f1dc4c4c9d997cca8d1b7fb01187a218aab932328247dcf5694a10102 \
    --hash=sha256:989899a85f0d9a57cebb508bd1f194cb52f0e3f7e22ac259f33d148d6422375c \
    --hash=sha256:9c02361af9bfa10ff1ccf967fc75159e56b1c8093caf565739ed77a559c1f29f \
    --hash=sha256:a0583076c4152b417a3a8a5d81ec02f58c09b67d3f22d5857e64c8734ceada8c \
    --hash=sha256:a1f4cbc70f6b7f0c939522118820e708e0d490edca42d852fa8004ec16780be2 \
    --hash=sha256:a690f2ff460682ea985e8d38ec541be97e0977fa0544aadc21efc116ff8d7579 \
    --hash=sha256:a75d45fbc20651c1b72e4111149fed3b33d270b0a4fb78328c54d965f28d55e1 \
    --hash=sha256:aedceb6090a60854b31bf9571dc55f642a3fa5b91f11b62bcef167c52cac93d8 \
    --hash=sha256:af91497dc098718e634d6ec8f91b182aea6bb3690f333fc9a7777bc70abe8810 \
    --hash=sha256:b4ab2b92a3e6ed552adbe9350fd2ef3aa0cc7853cf91569f9dbed0c0699bbeab \
    --hash=sha256:b8c7e514075696ca0f60c337f9e416e61d7ccbc1aa879a56c39181ed90ec3059 \
    --hash=sha256:bcd29945fafd12484cf37b1d84f12f0e7a33ba3eac5836531c6bd5283a6b3a0c \
    --hash=sha256:bfa550d9ae39e827a6e7198076df763571cb53397084974a6948af558355e028 \
    --hash=sha256:c3d68751668c66c7a140b6023dba5d5d507f72063407bb609d3a5b0f3b8dfbe4 \
    --hash=sha256:c69d5cad51388522b98a99b4be1b77316de85b0c0523fa865e0ea58bbb622e0a \
    --hash=sha256:c8eed5c015685106db15dd103fd040948ddca9197b1dd02222711815ea782a27 \
    --hash=sha256:cee29846471ce60226b18e931d8c1c66a158db94853e3e79bc2da9bd22345008 \
    --hash=sha256:d02f4ebe15aac7cdacce1a628e556c1983f26d140fd2e0ac5e0a090e605a2d38 \
    --hash=sha256:d566a4e09b8979be8ab9f843bac0dd216c81f5e5f45661a9b25cd162ed80508c \
    --hash=sha256:d80a7232938d523c1a12f6b1794ab5efb1ae77ad3fde79de4bb558d8ab261619 \
    --hash=sha256:d89a82937ce4037f092e9848a7bbcc65bc8e9fc9aef2bb74f5c15e7d21a73080 \
    --hash=sha256:da394654c6da15c1d37f0b7ec5afd325c69a15ceafee2afba14b67a5df8a82c8 \
    --hash=sha256:ddd1fe25af330f4e003421636746a546474e4ccd8f239f55d2898d80983d20ed \
    --hash=sha256:e6dd395d1a704e34a9fac00b25f0036dce6654c6b898be6f872ac2bb4f2eda48 \
    --hash=sha256:eeb6860b0f4bfa402de8929833fe5370fa34069c7ebacb2d543cb017f21fb891 \
    --hash=sha256:f3953d2f504176f929862e5579cfc421860c33e9707f585d70d24e1096accdf7 \
    --hash=sha256:f988f7f8164a6079c705c39e2d75dbe9967e3dacafe041420d9af7b9ee424162
    # via -r requirements-build.in
flit-core==3.10.1 \
    --hash=sha256:66e5b87874a0d6e39691f0e22f09306736b633548670ad3c09ec9db03c5662f7 \
    --hash=sha256:cb31a76e8b31ad3351bb89e531f64ef2b05d1e65bd939183250bf81ddf4922a8
    # via
    #   -r requirements-build.in
    #   flit-scm
flit-scm==1.7.0 \
    --hash=sha256:961bd6fb24f31bba75333c234145fff88e6de0a90fc0f7e5e7c79deca69f6bb2 \
    --hash=sha256:9e864caa8a63f708f5bb2f1b5b53eedcd4da75ec2cc6221a64cea7aa5c9eae1a
    # via -r requirements-build.in
hatch-fancy-pypi-readme==24.1.0 \
    --hash=sha256:26ec5c7cfd9f604eff0ae6c927d7b197b220706dca203f0aad1928abc81f3a46 \
    --hash=sha256:44dd239f1a779b9dcf8ebc9401a611fd7f7e3e14578dcf22c265dfaf7c1514b8
    # via -r requirements-build.in
hatch-vcs==0.4.0 \
    --hash=sha256:093810748fe01db0d451fabcf2c1ac2688caefd232d4ede967090b1c1b07d9f7 \
    --hash=sha256:b8a2b6bee54cf6f9fc93762db73890017ae59c9081d1038a41f16235ceaf8b2c
    # via -r requirements-build.in
hatchling==1.26.1 \
    --hash=sha256:3e8265ba6b7e86b71b5843e6e14b8a06a76d328ff1dda146e4e5aa8e5cd189a2 \
    --hash=sha256:8d956592e6c5d5df1f591be1e97a941c7b9ec6fd039f4c8624c29557bac034e9
    # via
    #   -r requirements-build.in
    #   hatch-fancy-pypi-readme
    #   hatch-vcs
packaging==24.2 \
    --hash=sha256:09abb1bccd265c01f4a3aa3f7a7db064b36514d2cba19a2f694fe6150451a759 \
    --hash=sha256:c228a6dc5e932d346bc5739379109d49e8853dd8223571c7c5b55260edc0b97f
    # via
    #   -r requirements-build.in
    #   hatchling
    #   setuptools-scm
pathspec==0.12.1 \
    --hash=sha256:a0d503e138a4c123b27490a4f7beda6a01c6f288df0e4a8b79c7eb0dc7b4cc08 \
    --hash=sha256:a482d51503a1ab33b1c67a6c3813a26953dbdc71c31dacaef9a838c4e29f5712
    # via
    #   -r requirements-build.in
    #   hatchling
pluggy==1.5.0 \
    --hash=sha256:2cffa88e94fdc978c4c574f15f9e59b7f4201d439195c3715ca9e2486f1d0cf1 \
    --hash=sha256:44e1ad92c8ca002de6377e165f3e0f1be63266ab4d554740532335b9d75ea669
    # via
    #   -r requirements-build.in
    #   hatchling
setuptools-scm[toml]==8.1.0 \
    --hash=sha256:42dea1b65771cba93b7a515d65a65d8246e560768a66b9106a592c8e7f26c8a7 \
    --hash=sha256:897a3226a6fd4a6eb2f068745e49733261a21f70b1bb28fce0339feb978d9af3
    # via
    #   -r requirements-build.in
    #   flit-scm
    #   hatch-vcs
tomli==2.0.2 \
    --hash=sha256:2ebe24485c53d303f690b0ec092806a085f07af5a5aa1464f3931eec36caaa38 \
    --hash=sha256:d46d457a85337051c36524bc5349dd91b1877838e2979ac5ced3e710ed8a60ed
    # via
    #   -r requirements-build.in
    #   flit-scm
    #   hatch-fancy-pypi-readme
    #   hatchling
    #   setuptools-scm
trove-classifiers==2024.10.21.16 \
    --hash=sha256:0fb11f1e995a757807a8ef1c03829fbd4998d817319abcef1f33165750f103be \
    --hash=sha256:17cbd055d67d5e9d9de63293a8732943fabc21574e4c7b74edf112b4928cf5f3
    # via
    #   -r requirements-build.in
    #   hatchling
typing-extensions==4.12.2 \
    --hash=sha256:04e5ca0351e0f3f85c6853954072df659d0d13fac324d0072316b67d7794700d \
    --hash=sha256:1a7ead55c7e559dd4dee8856e3a88b41225abfe1ce8df57b7c13915fe121ffb8
    # via
    #   -r requirements-build.in
    #   setuptools-scm
wheel==0.45.0 \
    --hash=sha256:52f0baa5e6522155090a09c6bd95718cc46956d1b51d537ea5454249edb671c7 \
    --hash=sha256:a57353941a3183b3d5365346b567a260a0602a0f8a635926a7dede41b94c674a
    # via -r requirements-build.in

# The following packages are considered to be unsafe in a requirements file:
setuptools==75.3.0 \
    --hash=sha256:f2504966861356aa38616760c0f66568e535562374995367b4e69c7143cf6bcd \
    --hash=sha256:fba5dd4d766e97be1b1681d98712680ae8f2f26d7881245f2ce9e40714f1a686
    # via
    #   -r requirements-build.in
    #   setuptools-scm

stackrox/operator/bundle_helpers on  master [$!?] via 🐍 v3.9.19 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ git st .
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   requirements-build.txt

no changes added to commit (use "git add" and/or "git commit -a")

stackrox/operator/bundle_helpers on  master [$!?] via 🐍 v3.9.19 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
❯ git diff requirements-build.txt
diff --git a/operator/bundle_helpers/requirements-build.txt b/operator/bundle_helpers/requirements-build.txt
index b1f7807835..15358a055b 100644
--- a/operator/bundle_helpers/requirements-build.txt
+++ b/operator/bundle_helpers/requirements-build.txt
@@ -4,10 +4,6 @@
 #
 #    pip-compile --allow-unsafe --generate-hashes requirements-build.in
 #
-calver==2022.6.26 \
-    --hash=sha256:a1d7fcdd67797afc52ee36ffb8c8adf6643173864306547bfd1380cbce6310a0 \
-    --hash=sha256:e05493a3b17517ef1748fbe610da11f10485faa7c416b9d33fd4a52d74894f8b
-    # via -r requirements-build.in
 cython==3.0.11 \
     --hash=sha256:0b1d1f6f94cc5d42a4591f6d60d616786b9cd15576b112bc92a23131fcf38020 \
     --hash=sha256:0e25f6425ad4a700d7f77cd468da9161e63658837d1bc34861a9861a4ef6346d \
@@ -79,6 +75,12 @@ cython==3.0.11 \
 flit-core==3.10.1 \
     --hash=sha256:66e5b87874a0d6e39691f0e22f09306736b633548670ad3c09ec9db03c5662f7 \
     --hash=sha256:cb31a76e8b31ad3351bb89e531f64ef2b05d1e65bd939183250bf81ddf4922a8
+    # via
+    #   -r requirements-build.in
+    #   flit-scm
+flit-scm==1.7.0 \
+    --hash=sha256:961bd6fb24f31bba75333c234145fff88e6de0a90fc0f7e5e7c79deca69f6bb2 \
+    --hash=sha256:9e864caa8a63f708f5bb2f1b5b53eedcd4da75ec2cc6221a64cea7aa5c9eae1a
     # via -r requirements-build.in
 hatch-fancy-pypi-readme==24.1.0 \
     --hash=sha256:26ec5c7cfd9f604eff0ae6c927d7b197b220706dca203f0aad1928abc81f3a46 \
@@ -119,12 +121,14 @@ setuptools-scm[toml]==8.1.0 \
     --hash=sha256:897a3226a6fd4a6eb2f068745e49733261a21f70b1bb28fce0339feb978d9af3
     # via
     #   -r requirements-build.in
+    #   flit-scm
     #   hatch-vcs
 tomli==2.0.2 \
     --hash=sha256:2ebe24485c53d303f690b0ec092806a085f07af5a5aa1464f3931eec36caaa38 \
     --hash=sha256:d46d457a85337051c36524bc5349dd91b1877838e2979ac5ced3e710ed8a60ed
     # via
     #   -r requirements-build.in
+    #   flit-scm
     #   hatch-fancy-pypi-readme
     #   hatchling
     #   setuptools-scm

stackrox/operator/bundle_helpers on  master [$!?] via 🐍 v3.9.19 via ❄️  impure (shell) on ☁️  mclasmei@redhat.com 
```

porridge commented:
The kind of dancing we have to do around the ~200 lines of Python here is unbelievable.
How about we rewrite `bundle_helpers` in Go rather than waste time on troubleshooting this discrepancy?

msugakov commented:
https://redhat-internal.slack.com/archives/C073B14UE10/p1737552077917849 related discussion.

msugakov commented on operator/bundle_helpers/requirements.in:
```diff
@@ -1,2 +1,2 @@
 pyyaml==6.0.2
-pytest==7.0.1
```
@mclasmeier `pytest` can be removed from this file because the one used in GHA and local builds has `pytest` (where it probably needs to be bumped).
https://github.com/stackrox/stackrox/blob/master/operator/bundle_helpers/requirements-gha.txt

msugakov commented:
This is likely to be replaced by https://github.com/stackrox/stackrox/pull/14257

Also see https://issues.redhat.com/browse/ROX-27677
